src/hashutil/mod.rs

```diff
@@ -0,0 +1,14 @@
+
+// TODO : Find proper hash libaray
+// Digest? hashes
+pub trait TempHashContainer {
```

Comment:
This one maybe:

https://docs.rs/digest/latest/digest/trait.Digest.html

Reply:
Although, it might not fit fully if you also want to capture:

https://github.com/starkware-libs/stone-prover/blob/main/src/starkware/channel/noninteractive_prover_felt_channel.cc